### PR TITLE
feat: adds regionality

### PR DESCRIPTION
**Pull request by Aayush-Abhyarthi**
### Description

Replace this text with a summary of the changes in this PR. Include why the changes are needed and context about the changes. List required dependencies. If there is a Git issue for the change, please link to it.

https://github.com/terraform-ibm-modules/terraform-ibm-cbr/issues/221

#### Changes that affect the core Terraform module or submodules
- [ ] Bug fix
- [ ] New feature
- [ ] Dependency update

#### Changes that don't affect the core Terraform module or submodules
- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other

#### Release required?
Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning).

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

File: cbr-service-profile/main.tf [hunk 1]
```diff
@@ -28,6 +28,7 @@ locals {
 
   vpc_zone_list = (length(var.zone_vpc_crn_list) > 0) ? [{
     name             = "${var.prefix}-cbr-vpc-zone"
+    region           = var.region_vpc_zone_list
```
Comment (vburckhardt) [inline cbr-service-profile/main.tf:31]: It's likely not needed for vpcs given that a vpc is already in a single region only. Region is only needed for the service list.

File: main.tf [hunk 2]
```diff
@@ -13,6 +13,7 @@ data "ibm_iam_account_settings" "iam_account_settings" {
 module "cbr_zone" {
   source             = "./cbr-zone-module"
   name               = var.name
+  region             = var.region
```
Comment (vburckhardt) [inline main.tf:16]: It looks like there is some code missing in this PR as the ./cbr-zone-module is not updated here to take the region as input.

**Review by vburckhardt — CHANGES_REQUESTED** [review]
Thank you for the PR. I put some comments inline. Also I think it would be desirable to update an example with the region and tests it out. @daniel-butler-irl can help on the testing once the region is updated.

File: main.tf (hunk 2 above)
Comment (Aayush-Abhyarthi) [reply main.tf:16]: Screenshot 2023-06-20 at 2 18 09 AM


Region does not seems to be an expected argument in the "cbr_zone" module

File: cbr-service-profile/main.tf (hunk 1 above)
Comment (Aayush-Abhyarthi) [reply cbr-service-profile/main.tf]: Updated

**Comment by vburckhardt** [comment]
The PR seems to be missing the logic that pass the region to the provider? How was the logic tested in this PR.

**Comment by vburckhardt** [comment]
The region is under the 'location' field  at https://github.com/terraform-ibm-modules/terraform-ibm-cbr/blob/99d4b2ed9d941c58a4b6461c9022451628d39a71/cbr-zone-module/variables.tf#L62 - so this value should be passed in the object at https://github.com/terraform-ibm-modules/terraform-ibm-cbr/blob/99d4b2ed9d941c58a4b6461c9022451628d39a71/cbr-service-profile/main.tf#L49 in the profile

**Comment by Aayush-Abhyarthi** [comment]
@daniel-butler-irl  as per the comment (https://github.com/terraform-ibm-modules/terraform-ibm-cbr/pull/240#pullrequestreview-1485790588) I need to update an example with region and test it.

**Review by vburckhardt — CHANGES_REQUESTED** [review]
I think there should be a check in the module to prevent setting the location field for services that do not support it. Here is a loop through all of the services.

module.cbr_account_level.module.cbr_zone["cloud-object-storage"].ibm_cbr_zone.cbr_zone: Modifying... [id=627340c5ae3d8626068d6c87441423cf]
module.cbr_account_level.module.cbr_zone["messages-for-rabbitmq"].ibm_cbr_zone.cbr_zone: Modifying... [id=627340c5ae3d8626068d6c8744143edb]
module.cbr_account_level.module.cbr_zone["databases-for-enterprisedb"].ibm_cbr_zone.cbr_zone: Modifying... [id=4a92e0ad3a7f8ddb78e50fbd9015f2ac]
module.cbr_account_level.module.cbr_zone["server-protect"].ibm_cbr_zone.cbr_zone: Modifying... [id=90589adcbd803505a2c7d61d1017a348]
module.cbr_account_level.module.cbr_zone["schematics"].ibm_cbr_zone.cbr_zone: Modifying... [id=91e84a60a1e96a1aba548d3e9116827a]
module.cbr_account_level.module.cbr_zone["event-notifications"].ibm_cbr_zone.cbr_zone: Modifying... [id=627340c5ae3d8626068d6c8744142fcf]
module.cbr_account_level.module.cbr_zone["directlink"].ibm_cbr_zone.cbr_zone: Modifying... [id=a9e8e9d3fda1927a2c0a804ad517ae03]
module.cbr_account_level.module.cbr_zone["databases-for-mysql"].ibm_cbr_zone.cbr_zone: Modifying... [id=90589adcbd803505a2c7d61d10179760]
module.cbr_account_level.module.cbr_zone["user-management"].ibm_cbr_zone.cbr_zone: Modifying... [id=627340c5ae3d8626068d6c87441448f0]
module.cbr_account_level.module.cbr_zone["iam-groups"].ibm_cbr_zone.cbr_zone: Modifying... [id=91e84a60a1e96a1aba548d3e91169a3e]
module.cbr_account_level.module.cbr_zone["databases-for-etcd"].ibm_cbr_zone.cbr_zone: Modifying... [id=91e84a60a1e96a1aba548d3e91168f7b]
module.cbr_account_level.module.cbr_zone["databases-for-mongodb"].ibm_cbr_zone.cbr_zone: Modifying... [id=f31c2047056e22987f156b8782168070]
module.cbr_account_level.module.cbr_zone["databases-for-postgresql"].ibm_cbr_zone.cbr_zone: Modifying... [id=91e84a60a1e96a1aba548d3e911665ff]
module.cbr_account_level.module.cbr_zone["messages-for-rabbitmq"].ibm_cbr_zone.cbr_zone: Modifications complete after 1s [id=627340c5ae3d8626068d6c8744143edb]
module.cbr_account_level.module.cbr_zone["messagehub"].ibm_cbr_zone.cbr_zone: Modifying... [id=f31c2047056e22987f156b8782167d9b]
module.cbr_account_level.module.cbr_zone["databases-for-enterprisedb"].ibm_cbr_zone.cbr_zone: Modifications complete after 1s [id=4a92e0ad3a7f8ddb78e50fbd9015f2ac]
module.cbr_account_level.module.cbr_zone["server-protect"].ibm_cbr_zone.cbr_zone: Modifications complete after 1s [id=90589adcbd803505a2c7d61d1017a348]
module.cbr_account_level.module.cbr_zone["secrets-manager"].ibm_cbr_zone.cbr_zone: Modifying... [id=4a92e0ad3a7f8ddb78e50fbd9015ec0f]
module.cbr_account_level.module.cbr_zone["cloud-object-storage"].ibm_cbr_zone.cbr_zone: Modifications complete after 1s [id=627340c5ae3d8626068d6c87441423cf]
module.cbr_account_level.module.cbr_zone["containers-kubernetes"].ibm_cbr_zone.cbr_zone: Modifying... [id=f31c2047056e22987f156b878216833c]
module.cbr_account_level.module.cbr_zone["apprapp"].ibm_cbr_zone.cbr_zone: Modifying... [id=4a92e0ad3a7f8ddb78e50fbd9015e982]
module.cbr_account_level.module.cbr_zone["event-notifications"].ibm_cbr_zone.cbr_zone: Modifications complete after 1s [id=627340c5ae3d8626068d6c8744142fcf]
module.cbr_account_level.module.cbr_zone["databases-for-elasticsearch"].ibm_cbr_zone.cbr_zone: Modifying... [id=a9e8e9d3fda1927a2c0a804ad517a38b]
module.cbr_account_level.module.cbr_zone["databases-for-mysql"].ibm_cbr_zone.cbr_zone: Modifications complete after 2s [id=90589adcbd803505a2c7d61d10179760]
module.cbr_account_level.module.cbr_zone["databases-for-redis"].ibm_cbr_zone.cbr_zone: Modifying... [id=91e84a60a1e96a1aba548d3e91165b69]
module.cbr_account_level.module.cbr_zone["databases-for-cassandra"].ibm_cbr_zone.cbr_zone: Modifying... [id=91e84a60a1e96a1aba548d3e91167304]
module.cbr_account_level.module.cbr_zone["schematics"].ibm_cbr_zone.cbr_zone: Modifications complete after 2s [id=91e84a60a1e96a1aba548d3e9116827a]
module.cbr_account_level.module.cbr_zone["is"].ibm_cbr_zone.cbr_zone: Modifying... [id=a9e8e9d3fda1927a2c0a804ad517b5b9]
module.cbr_account_level.module.cbr_zone["databases-for-postgresql"].ibm_cbr_zone.cbr_zone: Modifications complete after 1s [id=91e84a60a1e96a1aba548d3e911665ff]
module.cbr_account_level.module.cbr_zone["databases-for-mongodb"].ibm_cbr_zone.cbr_zone: Modifications complete after 2s [id=f31c2047056e22987f156b8782168070]
module.cbr_account_level.module.cbr_zone["codeengine"].ibm_cbr_zone.cbr_zone: Modifying... [id=a9e8e9d3fda1927a2c0a804ad517a67f]
module.cbr_account_level.module.cbr_zone["compliance"].ibm_cbr_zone.cbr_zone: Modifying... [id=a9e8e9d3fda1927a2c0a804ad517a435]
module.cbr_account_level.module.cbr_zone["messagehub"].ibm_cbr_zone.cbr_zone: Modifications complete after 1s [id=f31c2047056e22987f156b8782167d9b]
module.cbr_account_level.module.cbr_zone["secrets-manager"].ibm_cbr_zone.cbr_zone: Modifications complete after 1s [id=4a92e0ad3a7f8ddb78e50fbd9015ec0f]
module.cbr_account_level.module.cbr_zone["databases-for-redis"].ibm_cbr_zone.cbr_zone: Modifications complete after 0s [id=91e84a60a1e96a1aba548d3e91165b69]
module.cbr_account_level.module.cbr_zone["databases-for-cassandra"].ibm_cbr_zone.cbr_zone: Modifications complete after 0s [id=91e84a60a1e96a1aba548d3e91167304]
module.cbr_account_level.module.cbr_zone["is"].ibm_cbr_zone.cbr_zone: Modifications complete after 1s [id=a9e8e9d3fda1927a2c0a804ad517b5b9]
module.cbr_account_level.module.cbr_zone["codeengine"].ibm_cbr_zone.cbr_zone: Modifications complete after 1s [id=a9e8e9d3fda1927a2c0a804ad517a67f]
module.cbr_account_level.module.cbr_zone["databases-for-etcd"].ibm_cbr_zone.cbr_zone: Modifications complete after 3s [id=91e84a60a1e96a1aba548d3e91168f7b]
module.cbr_account_level.module.cbr_zone["apprapp"].ibm_cbr_zone.cbr_zone: Modifications complete after 2s [id=4a92e0ad3a7f8ddb78e50fbd9015e982]
module.cbr_account_level.module.cbr_zone["databases-for-elasticsearch"].ibm_cbr_zone.cbr_zone: Modifications complete after 2s [id=a9e8e9d3fda1927a2c0a804ad517a38b]
╷
│ Error: ReplaceZoneWithContext failed No service zones are found for the specified service reference's `location` property value 'eu-gb'.
│ {
│     "StatusCode": 400,
│     "Headers": {
│         "Akamai-Grn": [
│             "0.9e57dd58.1690538444.2b5da9e8"
│         ],
│         "Cache-Control": [
│             "no-cache,no-store"
│         ],
│         "Content-Length": [
│             "380"
│         ],
│         "Content-Type": [
│             "application/json; charset=utf-8"
│         ],
│         "Date": [
│             "Fri, 28 Jul 2023 10:00:44 GMT"
│         ],
│         "Expires": [
│             "Thursday, 1 January 1970 00:00:00 GMT"
│         ],
│         "Pragma": [
│             "no-cache"
│         ],
│         "Set-Cookie": [
│             "ak_bmsc=F4AAA170C668638FA90A3BE767A41314~000000000000000000000000000000~YAAQnlfdWGSc03CJAQAAjYfxmxSqm1FBf4/pXKhZEoZlTrC3rTKc8Wz6POoDmF92LlBKghPn4JSy+4+fZiDam2YB4J4X2J5MoJUlljcRBgqscna7sZxEXujfbwliJE7fr0Q0BROMN2iyWtphdI3yhZnvx4kc//YbJ0loRm7r7G8rSj503GarJ6gkytuG8UJxSsydxKtVWkp/iUm8pBwQre6Hq8NmA8Qd/7M9YfzCaMm6B/weszm5yX3uDln9Z3Xx0zNfONtSZgjRUpIZtO3k5GAKTA9DOgsi2v/iTtIPsOx+V/iupSe8hd4V7+CrjeLnoq3FxhK4nzcPxwC3AAl1HlxTCdUO0iQdP56s7RHQDUZ1ZTK0e0w2FdiMzgotRjrvd3s=; Domain=.cloud.ibm.com; Path=/; Expires=Fri, 28 Jul 2023 12:00:44 GMT; Max-Age=7200; HttpOnly"
│         ],
│         "Strict-Transport-Security": [
│             "max-age=31536000; includeSubDomains"
│         ],
│         "Transaction-Id": [
│             "c0a8bb84-d1fb-462e-b42b-42ed772ee094"
│         ],
│         "X-Content-Type-Options": [
│             "nosniff"
│         ],
│         "X-Correlation-Id": [
│             "c0a8bb84-d1fb-462e-b42b-42ed772ee094"
│         ],
│         "X-Proxy-Upstream-Service-Time": [
│             "116"
│         ],
│         "X-Ratelimit-Limit": [
│             "10"
│         ],
│         "X-Ratelimit-Remaining": [
│             "2"
│         ],
│         "X-Ratelimit-Reset": [
│             "1690538445"
│         ]
│     },
│     "Result": {
│         "errors": [
│             {
│                 "code": "no_service_zones_found_for_service_ref_property",
│                 "message": "No service zones are found for the specified service reference's `location` property value 'eu-gb'.",
│                 "target": {
│                     "name": "value",
│                     "type": "field",
│                     "value": "serviceName=compliance,accountId=7aa6f7b185f2e3170fac9919aa1769ee,location=eu-gb"
│                 }
│             }
│         ],
│         "status_code": 400,
│         "trace": "c0a8bb84-d1fb-462e-b42b-42ed772ee094"
│     },
│     "RawResult": null
│ }
│ 
│ 
│   with module.cbr_account_level.module.cbr_zone["compliance"].ibm_cbr_zone.cbr_zone,
│   on ../../cbr-zone-module/main.tf line 7, in resource "ibm_cbr_zone" "cbr_zone":
│    7: resource "ibm_cbr_zone" "cbr_zone" {
│ 
╵
╷
│ Error: ReplaceZoneWithContext failed No service zones are found for the specified service reference's `location` property value 'eu-gb'.
│ {
│     "StatusCode": 400,
│     "Headers": {
│         "Akamai-Grn": [
│             "0.9e57dd58.1690538442.2b5da42d"
│         ],
│         "Cache-Control": [
│             "no-cache,no-store"
│         ],
│         "Content-Length": [
│             "380"
│         ],
│         "Content-Type": [
│             "application/json; charset=utf-8"
│         ],
│         "Date": [
│             "Fri, 28 Jul 2023 10:00:42 GMT"
│         ],
│         "Expires": [
│             "Thursday, 1 January 1970 00:00:00 GMT"
│         ],
│         "Pragma": [
│             "no-cache"
│         ],
│         "Set-Cookie": [
│             "ak_bmsc=112F02177FC7BFD58EA97AF683C0874B~000000000000000000000000000000~YAAQnlfdWCGc03CJAQAA3H/xmxRhHlfN030b2nfuzkWfT0CbFWIowDhptQoN7qc2jvY1UWTajzegxEORI8+dceidY0d/m/JsUaNjZZGbrJqF8eyiYzmbuYBFm/oZKH4N2/BdDU0ESP4D88ef4vYlMmT5pQZ9aAAM1JIteBJbeeg+hxPIagAwbqlPMQfa92XRdCCs2SwfSRxOAln59uHUzZDfUGz9W1SNfLRj4LSGQHC91JReaj5yyAY+yYbek5fti/aCVvGFUK60I+y5se+W/xh0vxuuXYu9CnluuQ1XyhHp6sRkFLUZI7CAMZZMJ1ZE6e1uoH4bubYPDeGmbIXLgSnysIVjtvjM/hg5cQ1Pm+Js85w4wSJR1jHoUm1frqjGtb8=; Domain=.cloud.ibm.com; Path=/; Expires=Fri, 28 Jul 2023 12:00:42 GMT; Max-Age=7200; HttpOnly"
│         ],
│         "Strict-Transport-Security": [
│             "max-age=31536000; includeSubDomains"
│         ],
│         "Transaction-Id": [
│             "d8c9fc67-88c4-4f9e-8f6b-dd79004c9874"
│         ],
│         "X-Content-Type-Options": [
│             "nosniff"
│         ],
│         "X-Correlation-Id": [
│             "d8c9fc67-88c4-4f9e-8f6b-dd79004c9874"
│         ],
│         "X-Proxy-Upstream-Service-Time": [
│             "144"
│         ],
│         "X-Ratelimit-Limit": [
│             "10"
│         ],
│         "X-Ratelimit-Remaining": [
│             "6"
│         ],
│         "X-Ratelimit-Reset": [
│             "1690538443"
│         ]
│     },
│     "Result": {
│         "errors": [
│             {
│                 "code": "no_service_zones_found_for_service_ref_property",
│                 "message": "No service zones are found for the specified service reference's `location` property value 'eu-gb'.",
│                 "target": {
│                     "name": "value",
│                     "type": "field",
│                     "value": "serviceName=directlink,accountId=7aa6f7b185f2e3170fac9919aa1769ee,location=eu-gb"
│                 }
│             }
│         ],
│         "status_code": 400,
│         "trace": "d8c9fc67-88c4-4f9e-8f6b-dd79004c9874"
│     },
│     "RawResult": null
│ }
│ 
│ 
│   with module.cbr_account_level.module.cbr_zone["directlink"].ibm_cbr_zone.cbr_zone,
│   on ../../cbr-zone-module/main.tf line 7, in resource "ibm_cbr_zone" "cbr_zone":
│    7: resource "ibm_cbr_zone" "cbr_zone" {
│ 
╵
╷
│ Error: ReplaceZoneWithContext failed No service zones are found for the specified service reference's `location` property value 'eu-gb'.
│ {
│     "StatusCode": 400,
│     "Headers": {
│         "Akamai-Grn": [
│             "0.9e57dd58.1690538442.2b5da446"
│         ],
│         "Cache-Control": [
│             "no-cache,no-store"
│         ],
│         "Content-Length": [
│             "380"
│         ],
│         "Content-Type": [
│             "application/json; charset=utf-8"
│         ],
│         "Date": [
│             "Fri, 28 Jul 2023 10:00:42 GMT"
│         ],
│         "Expires": [
│             "Thursday, 1 January 1970 00:00:00 GMT"
│         ],
│         "Pragma": [
│             "no-cache"
│         ],
│         "Set-Cookie": [
│             "ak_bmsc=07392B9FC8539D634F1B55999A7AD506~000000000000000000000000000000~YAAQnlfdWCOc03CJAQAA9n/xmxQfSMWLglSAXdat8DJHY+SnqFRc7S//4Fr/huLMmQ+xBpWW/vbjIJp4AnK5DlPGNyR1c1ZK3NR09dpoDooHfyRSJ1Tl/HMPaXYKbu/MawBpfxAWOmJJ5u5lOD3CWJdPWpulhxcq3mJx5b4GbKfj235SbsXickb5FDUJ0JSi1ErJ8dG53tibR+DQjuLi+/BhBUKhUk6erp4IEKi9eFKdaT00ps5oRW0DmtvMCsuisA19HB+Xh2JZXjcArO55CW5GmWF6QSVlZzvOSDAMrd3iw47He+tghn9pBjsaBLy2NHdo6zFlY6Q+QGEGWKt5g8v1BBT+amqbxab8meYBB06Q3S8pg2PTgbFrlZ/7dCiToIw=; Domain=.cloud.ibm.com; Path=/; Expires=Fri, 28 Jul 2023 12:00:42 GMT; Max-Age=7200; HttpOnly"
│         ],
│         "Strict-Transport-Security": [
│             "max-age=31536000; includeSubDomains"
│         ],
│         "Transaction-Id": [
│             "da8e84b1-7c33-410d-8918-a591210067fd"
│         ],
│         "X-Content-Type-Options": [
│             "nosniff"
│         ],
│         "X-Correlation-Id": [
│             "da8e84b1-7c33-410d-8918-a591210067fd"
│         ],
│         "X-Proxy-Upstream-Service-Time": [
│             "192"
│         ],
│         "X-Ratelimit-Limit": [
│             "10"
│         ],
│         "X-Ratelimit-Remaining": [
│             "8"
│         ],
│         "X-Ratelimit-Reset": [
│             "1690538443"
│         ]
│     },
│     "Result": {
│         "errors": [
│             {
│                 "code": "no_service_zones_found_for_service_ref_property",
│                 "message": "No service zones are found for the specified service reference's `location` property value 'eu-gb'.",
│                 "target": {
│                     "name": "value",
│                     "type": "field",
│                     "value": "serviceName=iam-groups,accountId=7aa6f7b185f2e3170fac9919aa1769ee,location=eu-gb"
│                 }
│             }
│         ],
│         "status_code": 400,
│         "trace": "da8e84b1-7c33-410d-8918-a591210067fd"
│     },
│     "RawResult": null
│ }
│ 
│ 
│   with module.cbr_account_level.module.cbr_zone["iam-groups"].ibm_cbr_zone.cbr_zone,
│   on ../../cbr-zone-module/main.tf line 7, in resource "ibm_cbr_zone" "cbr_zone":
│    7: resource "ibm_cbr_zone" "cbr_zone" {
│ 
╵
╷
│ Error: ReplaceZoneWithContext failed No service zones are found for the specified service reference's `location` property value 'eu-gb'.
│ {
│     "StatusCode": 400,
│     "Headers": {
│         "Akamai-Grn": [
│             "0.9e57dd58.1690538444.2b5da5ed"
│         ],
│         "Cache-Control": [
│             "no-cache,no-store"
│         ],
│         "Content-Length": [
│             "391"
│         ],
│         "Content-Type": [
│             "application/json; charset=utf-8"
│         ],
│         "Date": [
│             "Fri, 28 Jul 2023 10:00:44 GMT"
│         ],
│         "Expires": [
│             "Thursday, 1 January 1970 00:00:00 GMT"
│         ],
│         "Pragma": [
│             "no-cache"
│         ],
│         "Set-Cookie": [
│             "ak_bmsc=32E26B2DF49AABD00BA034B07BF1D99A~000000000000000000000000000000~YAAQnlfdWFec03CJAQAADobxmxRUclJUKsbRe64z4HPo4J7rtkGlaO/up+Hmu8gnwLS+CqVLjVdjurmi4hlLyNL3COoB+vZUCdkSWpYXp63HpS+oHAzveJOJOzlzonC30jdwqnOQf0a11pB1ZDXAiHDC5v1o6r/IQfRy3GfeA4V3uRMqI07SMXnwJ7Q1u28DXImNousvymd7hnw3wVqROZOHCnN2kr2EELZb6scPNMziuym6awQIzVhURmzX8f6B34c3cUV4r2F2X3eeVPi3nTzf0ytupuhxZO35pEFanxjiBbw03mXDmxwbvCMYn8N72Q9F6hLhzofuodJYsCIZcL1MNP3a86hp8dbWOMnsR4fIz4s21sGfkyjVQQDvXc63kiI=; Domain=.cloud.ibm.com; Path=/; Expires=Fri, 28 Jul 2023 12:00:44 GMT; Max-Age=7200; HttpOnly"
│         ],
│         "Strict-Transport-Security": [
│             "max-age=31536000; includeSubDomains"
│         ],
│         "Transaction-Id": [
│             "4332c92c-38d3-4c15-b74a-30d584a4fac6"
│         ],
│         "X-Content-Type-Options": [
│             "nosniff"
│         ],
│         "X-Correlation-Id": [
│             "4332c92c-38d3-4c15-b74a-30d584a4fac6"
│         ],
│         "X-Proxy-Upstream-Service-Time": [
│             "115"
│         ],
│         "X-Ratelimit-Limit": [
│             "10"
│         ],
│         "X-Ratelimit-Remaining": [
│             "9"
│         ],
│         "X-Ratelimit-Reset": [
│             "1690538445"
│         ]
│     },
│     "Result": {
│         "errors": [
│             {
│                 "code": "no_service_zones_found_for_service_ref_property",
│                 "message": "No service zones are found for the specified service reference's `location` property value 'eu-gb'.",
│                 "target": {
│                     "name": "value",
│                     "type": "field",
│                     "value": "serviceName=containers-kubernetes,accountId=7aa6f7b185f2e3170fac9919aa1769ee,location=eu-gb"
│                 }
│             }
│         ],
│         "status_code": 400,
│         "trace": "4332c92c-38d3-4c15-b74a-30d584a4fac6"
│     },
│     "RawResult": null
│ }
│ 
│ 
│   with module.cbr_account_level.module.cbr_zone["containers-kubernetes"].ibm_cbr_zone.cbr_zone,
│   on ../../cbr-zone-module/main.tf line 7, in resource "ibm_cbr_zone" "cbr_zone":
│    7: resource "ibm_cbr_zone" "cbr_zone" {
│ 
╵
╷
│ Error: ReplaceZoneWithContext failed No service zones are found for the specified service reference's `location` property value 'eu-gb'.
│ {
│     "StatusCode": 400,
│     "Headers": {
│         "Akamai-Grn": [
│             "0.9e57dd58.1690538442.2b5da437"
│         ],
│         "Cache-Control": [
│             "no-cache,no-store"
│         ],
│         "Content-Length": [
│             "385"
│         ],
│         "Content-Type": [
│             "application/json; charset=utf-8"
│         ],
│         "Date": [
│             "Fri, 28 Jul 2023 10:00:42 GMT"
│         ],
│         "Expires": [
│             "Thursday, 1 January 1970 00:00:00 GMT"
│         ],
│         "Pragma": [
│             "no-cache"
│         ],
│         "Set-Cookie": [
│             "ak_bmsc=3ED444396AA0C35EC4F11D4815DEB6CD~000000000000000000000000000000~YAAQnlfdWCKc03CJAQAA6H/xmxSJxbvFVqzcIYOwbLGelzEAU17gSg2VwtOgOujqIyJX9c7MYh1MCcI6ppLnQsg7NfODvVP15uHw6/Ik/gIaKrR+DL2T8ik+yc902VB0CSg1GJDKQ3UsMRc4Mc0EBkjiA3n2rre6VGBSx+jnixYn77s/oFqB8qCoI0yFIkLOD4YUyzENL6Yvps1KDGb/CiV2YMyNugFUuxn4ciNK7ElB+qZVjDrcdIROgyV9XLwQ5MrYXFhmDNKUua4m8cPrU4MOOZhZX8+rscaoCT2kl0bV1dNBQry8oRgteRZ2GOkBZxJsWcB5tWd/QvCSRCWpa1Vv0rXB93xcD+0Bo5bfJOQ+ZkEXgOP2rFMdDCKGqVb7Y9U=; Domain=.cloud.ibm.com; Path=/; Expires=Fri, 28 Jul 2023 12:00:42 GMT; Max-Age=7200; HttpOnly"
│         ],
│         "Strict-Transport-Security": [
│             "max-age=31536000; includeSubDomains"
│         ],
│         "Transaction-Id": [
│             "38eff885-f3cf-4b7f-baad-a12de33106dd"
│         ],
│         "X-Content-Type-Options": [
│             "nosniff"
│         ],
│         "X-Correlation-Id": [
│             "38eff885-f3cf-4b7f-baad-a12de33106dd"
│         ],
│         "X-Proxy-Upstream-Service-Time": [
│             "120"
│         ],
│         "X-Ratelimit-Limit": [
│             "10"
│         ],
│         "X-Ratelimit-Remaining": [
│             "5"
│         ],
│         "X-Ratelimit-Reset": [
│             "1690538443"
│         ]
│     },
│     "Result": {
│         "errors": [
│             {
│                 "code": "no_service_zones_found_for_service_ref_property",
│                 "message": "No service zones are found for the specified service reference's `location` property value 'eu-gb'.",
│                 "target": {
│                     "name": "value",
│                     "type": "field",
│                     "value": "serviceName=user-management,accountId=7aa6f7b185f2e3170fac9919aa1769ee,location=eu-gb"
│                 }
│             }
│         ],
│         "status_code": 400,
│         "trace": "38eff885-f3cf-4b7f-baad-a12de33106dd"
│     },
│     "RawResult": null
│ }
│ 
│ 
│   with module.cbr_account_level.module.cbr_zone["user-management"].ibm_cbr_zone.cbr_zone,
│   on ../../cbr-zone-module/main.tf line 7, in resource "ibm_cbr_zone" "cbr_zone":
│    7: resource "ibm_cbr_zone" "cbr_zone" {
│ 
╵

File: cbr-service-profile/main.tf [hunk 3]
```diff
@@ -48,6 +48,7 @@ locals {
           ref = {
             account_id   = data.ibm_iam_account_settings.iam_account_settings.account_id
             service_name = serviceref
+            location     = (serviceref == "compliance" || serviceref == "directlink" || serviceref == "iam-groups" || serviceref == "containers-kubernetes" || serviceref == "user-management") ? null : var.location
```
Comment (vburckhardt) [inline cbr-service-profile/main.tf:51]: I think this should be a validation block to return an explicit error message to the caller if location is set for a service not supporting location.

The current logic would silently ignore the location field which may make call think that there is a restriction per location when in fact there is not, which is really dangerous from a security perspective.

**Review by vburckhardt — CHANGES_REQUESTED** [review]
Thanks - this is moving into the right direction. I put a comment inline. Could we also add the logic to the newly release fscloud profile module?

**Review by vburckhardt — APPROVED** [review]

**Pull request merged**